server/src/test/java/umm3601/user/UserControllerSpec.java

```diff
@@ -210,6 +216,34 @@ void canGetUsersWithAge37() throws IOException {
     assertEquals(2, userArrayListCaptor.getValue().size());
   }
 
+  @Test
+  void canGetUsersWithAge37Redux() throws JsonMappingException, JsonProcessingException {
```

Comment:
I am inclined to either include both and make it clear why they are both there or bring the argument matcher in to the one that doesn't have it and talk about the different ways of checking. I disagree that it would likely lead to students doing more testing than is needed. I don't think they use the presence of a test as a cue... they look at their coverage and what they need to test. I agree that removing the extra test classes we had was a good move, but I think multiple examples could serve them well.

Reply:
I've seen _numerous_ occasions where students copy "unnecessary" tests through all the way to their final projects, even when they're labelled as "just examples". In the heat of battle there's a lot of copy/paste, and when there's limited understanding folks tend to just copy and tweak.

I do agree that a variety of examples can be useful, though, and I don't want to discourage that. I just don't want to do it in a way that ends up cluttering up the code down the road. Maybe if we have multiple versions in Lab 4 but _not_ in the iteration template that would be better? I'm not sure.

If we do include both, then we definitely want to comment the heck out of them both and make it really clear that they're effectively redundant and that people should _not_ in general have multiple tests of the same thing like this.

I'm also fine with copying the argument matcher into the one we keep if you think that would be useful.

Maybe we should go through all the tests and try to identify key features like argument matchers that we want to illustrate somewhere. Then we could actually list the features, along with which test(s) highlight/use that feature. That might provide some structure and help students see these more as a resource to learn from and less as a pile of copy to mechanically copy and tweak?

Reply:
I'm inclined to make an issue about this and still go ahead and include them both in the code for this year. I really want to get this out the door, and I am willing to deal with whatever comes our way this semester as a result of including both. I agree that pointing out great elements of all of the tests and being really careful about what to include would be wise, but I also think that including it as it is and then having conversations about it will be fine for this year. I don't want this to hold up the whole PR, and I'm not in a position to fix everything this semester. I've tackled quite a few issues and made things a lot more consistent between the iteration template and the labs than it was. I'm not going to do the major overhaul this term, even though it might be a good thing.